compiler/ilgen/VirtualMachineOperandStack.cpp

```diff
@@ -130,6 +131,18 @@ VirtualMachineOperandStack::Commit(TR::IlBuilder *b)
       }
    }
 
+// Update the OperandStack_base and _stackTopRegister after the Virtual Machine moves the stack.
+// This call will normally be followed by a call to Reload if any of the stack values changed in the move
+void
+VirtualMachineOperandStack::UpdateStack(TR::IlBuilder *b, TR::IlValue *stack)
+   {
+   b->Store("OperandStack_base", stack);
+   // Adjust the vm _stackTopRegister by number of elements that have been pushed onto the stack.
+   // _stackTop is -1 at 0 pushes, 0 for 1 push, so # of elements to adjust by is _stackTop+1
+   _stackTopRegister->Store(b, stack);
+   _stackTopRegister->Adjust(b, (_stackTop+1)*_pushAmount);
```

Comment:
I don't think we need to adjust _stackTopRegister here, after John's change to remove all interim adjustments to the stack top register. _stackTop doesn't change with UpdateStack(), and Commit() will update _stackTopRegister in exactly the same way you have here (you clearly copied the lines from Commit :) ) when the next Commit() call is made.